addons/web_editor/static/src/js/editor/odoo-editor/test/spec/editor.test.js

```diff
@@ -3830,6 +3830,88 @@ X[]
                     contentAfter: '<p><br></p><p>[]<br></p>',
                 });
             });
+            it('preserve selected table row and insert paragraph in cell when whole table is not selected and enter key is pressed', async () => {
+                // Forward selection
+                await testEditor(BasicEditor, {
+                    contentBefore: unformat(
+                        `<table class="o_selected_table">
+                            <tbody>
+                                <tr><td class="o_selected_td"><p>[<br></p></td><td class="o_selected_td"><p><br></p></td><td class="o_selected_td"><p><br>]</p></td></tr>
```

Comment:
Do you have to put the classes? Doesn't the editor do it on its own when its sets the selection? Ideally you'd have `contentBefore` without the classes, and `contentBeforeEdit` would be what you wrote here.

Reply:
do you mean i need to use `contentBeforeEdit`  here?

Reply:
Yes. `contentBefore` without the `o_selected_*` classes, `contentBeforeEdit` with the `o_selected_*` classes.